File: appointments-service/src/appointments/dto/create-appointment.dto.ts
import { IsISO8601, IsOptional, IsString, MinLength } from 'class-validator';

export class CreateAppointmentDto {
  @IsString()
  @MinLength(1)
  patientId: string;

  @IsString()
  @MinLength(1)
  dentistId: string;

  @IsISO8601()
  startAt: string;

  @IsISO8601()
  endAt: string;

  @IsOptional()
  @IsString()
  reason?: string;

  @IsOptional()
  @IsString()
  notes?: string;
}

File: appointments-service-csharp/Dentia.Appointments.Api/Application/DTOs/CreateAppointmentDto.cs
using System.ComponentModel.DataAnnotations;

namespace Dentia.Appointments.Api.Application.DTOs;

public class CreateAppointmentDto
{
    [Required]
    [MinLength(1)]
    public string PatientId { get; set; } = string.Empty;

    [Required]
    [MinLength(1)]
    public string DentistId { get; set; } = string.Empty;

    [Required]
    public DateTime StartAt { get; set; }

    [Required]
    public DateTime EndAt { get; set; }

    public string? Reason { get; set; }

    public string? Notes { get; set; }
}